AppIntegrity/README.md

```diff
@@ -0,0 +1,53 @@
+#  App Integrity
+
+## Package description
+
+This package contains the logic to implement app integrity checks using [Firebase App Check](https://firebase.google.com/docs/app-check). It therefore encorporates the Firebase SDK as a dependency, along with the Networking module to enable backend requests.
+
+## Overview
+
+### What is app integrity?
+
+App Integrity describes checks made (in this case, by Firebase AppCheck) to verify that a genuine instance of your app is accessing your backend resources and that the device accessing it has not been tampered with.
+
+### Why do we need it for mobile?
+
+As a public client of STS, our mobile app is susceptible to malicious attacks from unauthorized clients. We have therefore implemented app integrity checks when the `/token` endpoint is called to ensure that:
+1. the access token is being issued to the genuine instance of the client requesting it, i.e. the client is not being impersonated
+
+2. that the authorization code has not been intercepted ([PKCE](https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow-with-pkce)).
+
+This therefore adds a layer of security to our authorization flow.
+
+## Testing
+
+### Testing locally
+
+Using the AppCheckDebugProviderFactory type when the build configuration is in Debug  allows us to pass a known debug token as an environment variable to Firebase App Check.
+
+This can be done by editing the scheme on xcode and passing an environment variable with the name FIRAAppCheckDebugToken.
+
+The value should be saved in an encrypted secrets file outside of the repository and passed in by including the file in your build config as shown below:
```

Comment:
The file itself should be encrypted but should come from your ["CI system's secure key store"](https://firebase.google.com/docs/app-check/ios/debug-provider)